pkg/core/interop_neo.go

```diff
@@ -600,6 +603,33 @@ func (ic *interopContext) contractMigrate(v *vm.VM) error {
 	return ic.contractDestroy(v)
 }
 
+// secp256k1Recover recovers speck256k1 public key.
+func (ic *interopContext) secp256k1Recover(v *vm.VM) error {
+	return ic.eccRecover(btcec.S256(), v)
+}
+
+// secp256r1Recover recovers speck256r1 public key.
+func (ic *interopContext) secp256r1Recover(v *vm.VM) error {
+	return ic.eccRecover(elliptic.P256(), v)
+}
+
+// eccRecover recovers public key using ECCurve set
+func (ic *interopContext) eccRecover(curve elliptic.Curve, v *vm.VM) error {
+	rBytes := v.Estack().Pop().Bytes()
+	sBytes := v.Estack().Pop().Bytes()
+	r := new(big.Int).SetBytes(rBytes)
```

Comment:
Most likely, it should be `emit.BytesToInt`. C#'s serialization differs from that of Go.

Reply:
Probably not, I'd expect these to be the same as ones used by `(*PublicKey).Verify`. See also [cross-chain contract](https://github.com/neo-ngd/CrossChain-NeoContract/blob/eba9397834cab6ee8f98968ccd21e0a44a667030/CCMC/CCMC.cs#L511).

Reply:
It'd actually be way better to have full signature as an argument, but we have what we have.